src/anaconda_auth/_conda/plugins.py

```diff
@@ -5,24 +5,77 @@
 
 """
 
+from frozendict import frozendict
+import sys
+
 from typing import Iterable
 from typing import Optional
 
 from conda import plugins
 from conda.plugins.types import CondaAuthHandler
+from conda.plugins.types import CondaPreCommand
 from conda.plugins.types import CondaSubcommand
 
 from anaconda_auth._conda.auth_handler import AnacondaAuthHandler
+from anaconda_auth._conda.auth_handler import TOKEN_DOMAIN_MAP
 from anaconda_auth._conda.conda_token import cli
+from anaconda_auth.config import AnacondaAuthConfig
+from anaconda_auth.config import AnacondaAuthSitesConfig
+
 
-__all__ = ["conda_subcommands", "conda_auth_handlers"]
+__all__ = ["conda_subcommands", "conda_auth_handlers", "conda_pre_commands"]
 
 
 def _cli_wrapper(argv: Optional[list[str]] = None) -> int:  # type: ignore
     # If argv is empty tuple, we need to set it back to None
     return cli(argv=argv or None)
 
 
+def merge_auth_configs(command: str) -> None:
```

Comment:
#172 has this code in `config.py`, but I do not agree it should be there; that module knows nothing about Conda and should stay that way